### PR TITLE
PA-239 edit draft apprenticeship

### PR DESCRIPTION
**Pull request by Pauljgraham**

### CODE REVIEW

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Controllers/DraftApprenticeshipControllerTests/WhenSelectingCourseOnEditApprenticeship.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        public GetCohortResponse Cohort;
+        public GetTrainingCoursesQueryResponse TrainingCourseResponse;
+        public Mock<ICommitmentsApiClient> CommitmentsApiClientMock;
+
+        public WhenSelectingCourseOnEditApprenticeshipFixture()
+        {
+            var fixture = new Fixture();
+            ViewModel = fixture.Create<SelectCourseViewModel>();
+            Request = fixture.Create<DraftApprenticeshipRequest>();
+            DraftApprenticeship = fixture.Build<EditDraftApprenticeshipViewModel>().Without(x => x.BirthDay).Without(x => x.BirthMonth).Without(x => x.BirthYear)
```
Comment (benagain) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Controllers/DraftApprenticeshipControllerTests/WhenSelectingCourseOnEditApprenticeship.cs:91]: Is it important to the tests that these properties are not set?

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Mappers/DraftApprenticeshipTests/WhenIMapEditDraftApprenticeshipDetailsToViewModel.cs [hunk 2]
```diff
@@ -197,5 +197,21 @@ public async Task ThenDeliveryModelIsMappedCorrectly(DeliveryModel dm)
             var result = await _act();
             Assert.AreEqual(dm, result.DeliveryModel);
         }
+
+        //[Test]
```
Comment (benagain) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Mappers/DraftApprenticeshipTests/WhenIMapEditDraftApprenticeshipDetailsToViewModel.cs:201]: Please can we remove commented code

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Mappers/WhenIMapEditDraftApprenticeshipToBaseDraftApprenticeshipRequest.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    public class WhenIMapEditDraftApprenticeshipToBaseDraftApprenticeshipRequest
+    {
+        private EditDraftApprenticeshipToBaseDraftApprenticeshipRequestMapper _mapper;
+        private EditDraftApprenticeshipViewModel _source;
+
+        [SetUp]
+        public void Arrange()
+        {
+            var fixture = new Fixture();
+            _source = fixture.Build<EditDraftApprenticeshipViewModel>().Without(x => x.BirthDay).Without(x => x.BirthMonth).Without(x => x.BirthYear)
```
Comment (benagain) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Mappers/WhenIMapEditDraftApprenticeshipToBaseDraftApprenticeshipRequest.cs:20]: Is it important that these properties are null?  I can't see any reason for it - but could be missing something!

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Models/DraftApprenticeshipViewModel.cs [hunk 4]
```diff
@@ -8,12 +8,12 @@ namespace SFA.DAS.ProviderCommitments.Web.Models
...
     public class DraftApprenticeshipViewModel
     {
-        public DraftApprenticeshipViewModel(DateTime? dateOfBirth, DateTime? startDate, DateTime? endDate) : base()
+        public DraftApprenticeshipViewModel(DateTime? dateOfBirth, DateTime? startDate, DateTime? endDate, DateTime? employmentEndDate = null) : base()
         {
             DateOfBirth = dateOfBirth == null ? new DateModel() : new DateModel(dateOfBirth.Value);
             StartDate = startDate == null ? new MonthYearModel("") : new MonthYearModel($"{startDate.Value.Month}{startDate.Value.Year}");
             EndDate = endDate == null ? new MonthYearModel("") : new MonthYearModel($"{endDate.Value.Month}{endDate.Value.Year}");
-            EmploymentEndDate = endDate == null ? new MonthYearModel("") : new MonthYearModel($"{endDate.Value.Month}{endDate.Value.Year}");
+            EmploymentEndDate = employmentEndDate == null ? new MonthYearModel("") : new MonthYearModel($"{endDate.Value.Month}{endDate.Value.Year}");
```
Comment (benagain) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Models/DraftApprenticeshipViewModel.cs:16]: Good catch!

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Controllers/DraftApprenticeshipControllerTests/WhenSelectingCourseOnEditApprenticeship.cs (hunk 1 above)
Comment (Pauljgraham) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Controllers/DraftApprenticeshipControllerTests/WhenSelectingCourseOnEditApprenticeship.cs:91]: Autofixture, creates them with invalid values, if not excluded

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Mappers/WhenIMapEditDraftApprenticeshipToBaseDraftApprenticeshipRequest.cs (hunk 3 above)
Comment (Pauljgraham) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Mappers/WhenIMapEditDraftApprenticeshipToBaseDraftApprenticeshipRequest.cs:20]: Autofixture, creates them with invalid values, if not excluded

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Mappers/DraftApprenticeshipTests/WhenIMapEditDraftApprenticeshipDetailsToViewModel.cs (hunk 2 above)
Comment (Pauljgraham) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Mappers/DraftApprenticeshipTests/WhenIMapEditDraftApprenticeshipDetailsToViewModel.cs:201]: done

**Review by benagain — APPROVED** [review]

**Pull request merged**